app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,60 @@
+class ArticlesController < ApplicationController
+  before_action :find_article, except: %i[index new create]
+
+  http_basic_authenticate_with name: 'user', password: 'secret', except: %i[index show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show
+    @article 
+  end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    @article = Article.new(article_params)
+    begin
+      @article.save!
+    rescue StandardError => e
```

Comment:
You can delete the error object if you are not going to use it.

Reply:
Is correct to put the error message in the console? 
```
rescue ActiveRecord::RecordInvalid => error
      puts error.message
      render :new 
```

Reply:
The correct way to do it is to use the Rails Logger in case there is the need to log the error.

Reply:
Got it, useful to debug  [Ruby Debug](https://guides.rubyonrails.org/debugging_rails_applications.html#the-logger)